packages/health/lib/src/health_factory.dart

```diff
@@ -348,23 +376,83 @@ class HealthFactory {
       throw HealthException(dataType, 'Not available on platform $_platformType');
     }
 
-    return await _dataQuery(startTime, endTime, dataType);
+    return await _dataQuery(startTime, endTime, dataType, isPriorityQueue);
   }
 
+  List<DataPointInput> requests = [];
+
   /// The main function for fetching health data
-  Future<List<HealthDataPoint>> _dataQuery(DateTime startTime, DateTime endTime, HealthDataType dataType) async {
-    final args = <String, dynamic>{
-      'dataTypeKey': dataType.name,
-      'dataUnitKey': _dataTypeToUnit[dataType]!.name,
-      'startTimeSec': startTime.millisecondsSinceEpoch ~/ 1000,
-      'endTimeSec': endTime.millisecondsSinceEpoch ~/ 1000,
-    };
-    List<Map>? fetchedDataPoints = await _channel.invokeListMethod('getData', args);
-    if (fetchedDataPoints != null) {
-      return _parse(dataType: dataType, dataPoints: fetchedDataPoints);
+  Future<List<HealthDataPoint>> _dataQuery(
+      DateTime startTime, DateTime endTime, HealthDataType dataType, bool isPriorityQueue) async {
+    Completer<List<HealthDataPoint>> _completer = new Completer();
+
+    final hours = _divideDateRangeIntoHours(startTime, endTime);
+    if (hours.isEmpty) {
```

Comment:
couldn't the _divideDataRangeIntoHours do this default value?

Reply:
It could, but then it stops being just divide function. I think it is more logical that way.

Reply:
Divide can return a single item.... I think it encapsulates everything better.